### PR TITLE
Megatron_LM checkpoint to NeMo checkpoint support 

### PR DESCRIPTION
**Pull request by yidong72**
# What does this PR do ?

Convert Megatron LM checkpoint to NeMo checkpoint. Copy over the weights, optimizer states, LR scheduler states.
Fix the issue of weight decay parameter group assignment. 
Tested that the script converts the old Megatron GPT and BERT checkpoint to `.nemo` file fine.

Tested with 40B checkpoint file, the NeMo Megatron can resume from it and continue the training. Need to verify the training curve works fine.

### CODE REVIEW

**Review by ericharper — DISMISSED** [review]
LGTM. Thanks!

**Review by ericharper — CHANGES_REQUESTED** [review]
Still testing if training is working with this PR.

**Review by ericharper — APPROVED** [review]
LGTM.
